### PR TITLE
feat(core): critical operations in UsersManager

### PR DESCRIPTION
**Pull request by sarkapalkovicova**
Critical operations in perun-roles.yml were marked.

### CODE REVIEW

File: perun-base/src/main/resources/perun-roles.yml [hunk 1]
```diff
@@ -5095,50 +5103,71 @@ perun_policies:
...
+    mfa_rules:
+      - MFA: User
 
   deleteUser_User_policy:
     policy_roles: []
     include_policies:
       - default_policy
+    mfa_rules:
+      - MFA: User
+      - MFA: Member
```
Comment (xflord) [inline perun-base/src/main/resources/perun-roles.yml:5129]: I think Member is affected only in the forceDelete version of this method (so the policy after this one). Also, user external sources are deleted in this method, so MFA: UserExtSource should be here imo. What do you think @Johaney-s @cuadradek ?

File: perun-base/src/main/resources/perun-roles.yml [hunk 2]
```diff
@@ -5095,50 +5103,71 @@ perun_policies:
...
 
   anonymizeUser_User_boolean_policy:
     policy_roles: []
     include_policies:
       - default_policy
+    mfa_rules:
+      - MFA: User
+      - MFA: Member
```
Comment (xflord) [inline perun-base/src/main/resources/perun-roles.yml:5145]: Same deal with the UserExtSources here

File: perun-base/src/main/resources/perun-roles.yml [hunk 3]
```diff
@@ -5204,22 +5233,33 @@ perun_policies:
     policy_roles: []
     include_policies:
       - default_policy
+    mfa_rules:
+      - MFA: User
```
Comment (xflord) [inline perun-base/src/main/resources/perun-roles.yml:5237]: I feel like whenever working with UES, we should check both the UES and User associated (so in `updateUserExtSource_UserExtSource_policy` as well) @Johaney-s  @cuadradek thoughts?

File: perun-base/src/main/resources/perun-roles.yml [hunk 4]
```diff
@@ -5412,81 +5452,107 @@ perun_policies:
...
+    mfa_rules:
+      -MFA: User
 
   deletePassword_String_String_policy:
     policy_roles:
       - REGISTRAR:
     include_policies:
       - default_policy
+    mfa_rules:
+      -MFA
```
Comment (xflord) [inline perun-base/src/main/resources/perun-roles.yml:5538]:
```suggestion
      -MFA:
```

**Review by xflord — COMMENTED** [review]
Again, these reviews can be a bit tricky, if you disagree with anything, comment here or mention it at the daily meeting

File: perun-base/src/main/resources/perun-roles.yml (hunk 3 above)
Comment (Johaney-s) [reply perun-base/src/main/resources/perun-roles.yml:5239]: here I agree; when deleting user I'm not so sure, but better make it stricter now and if needed we'll remove it later... so okay by me

**Review by xflord — APPROVED** [review]

**Review by Johaney-s — APPROVED** [review]

**Pull request merged**